File: DoubanFM/DBFMClient.m
//
//  DBFMClient.m
//  DoubanFM
//
//  Created by cc on 16/3/28.
//  Copyright © 2016年 cc. All rights reserved.
//

#import "DBFMClient.h"

@implementation User

- (void)updateWithDict:(id)json {
    self.user_id = json[@"user_id"];
    self.token = json[@"token"];
    self.expire = json[@"expire"];
    self.user_name = json[@"user_name"];
    self.email = json[@"email"];
}

@end


@implementation DBFMClient
+ (DBFMClient *)sharedClient {
    static DBFMClient *_self = NULL;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        _self = [[DBFMClient alloc] init];
    });
    return _self;
}

- (instancetype)init {
    if (self = [super init]) {
        _user = [[User alloc] init];
    }
    return self;
}


@end

File: DoubanFM/DBFMClient.swift
//
//  DBFMClient.swift
//  DoubanFM
//
//  Created by cc on 16/3/28.
//  Copyright © 2016年 cc. All rights reserved.
//

import Cocoa
import SwiftyJSON

class User: NSObject {
    var user_id: String?
    var token: String?
    var expire: String?
    var user_name: String?
    var email: String?
    
    func updateWithJSON(dict: JSON) {
        self.user_id = dict["user_id"].string
        self.token = dict["token"].string
        self.expire = dict["expire"].string
        self.user_name = dict["user_name"].string
        self.email = dict["email"].string
    }
    
    override func isEqual(object: AnyObject?) -> Bool {
        if let other = object as? User {
            return self.email == other.email
        }
        return false
    }
}

class DBFMClient: NSObject {
    static let sharedClient = DBFMClient()
    var user: User!
    var haveLogin: Bool {
        get {
            return self.user.user_id != nil
        }
    }
    
    override init() {
        super.init()
        user = User()
    }
}

let UserLoginSuccessNotification = "UserLoginSuccessNotification"
